beacon2-import.py

```diff
@@ -0,0 +1,641 @@
+
+    return parser.parse_args()
+
+
+def set_up_logging(verbosity: int):
+    """
+    Configures the logger for this script
+
+        Parameters:
+            verbosity (int):
+
+        Returns:
+            Nothing.
+    """
+
+    # configure log level to match the given verbosity
+    if verbosity > 1:
+        logging.basicConfig(level=logging.DEBUG)
+    elif verbosity == 1:
+        logging.basicConfig(level=logging.INFO)
+    else:
+        logging.basicConfig(level=logging.WARN)
+
+
+def string_as_bool(string: Any) -> bool:
+    """
+    Returns the bool value corresponding to the given value
+
+    used to typesave the api responses
+    """
+    if str(string).lower() in ("true", "yes", "on", "1"):
+        return True
+    else:
+        return False
+
+
+def set_up_galaxy_instance(galaxy_url: str, galaxy_key: str) -> GalaxyInstance:
+    """
+    Returns a galaxy instance with the given URL and api key.
+    Exits immediately if either connection or authentication to galaxy fails.
+
+        Parameters:
+            galaxy_url (str): Base URL of a galaxy instance
+            galaxy_key (str): API key with admin privileges
+
+        Returns:
+            gi (GalaxyInstance): Galaxy instance with confirmed admin access to the given galaxy instance
+    """
+
+    logging.info(f"trying to connect to galaxy at {galaxy_url}")
+
+    # configure a galaxy instance with galaxy_url and api_key
+    try:
+        gi = GalaxyInstance(galaxy_url, key=galaxy_key)
+    except Exception as e:
+        # if galaxy_url does not follow the scheme <protocol>://<host>:<port>, GalaxyInstance attempts guessing the URL
+        # this exception is thrown when neither "http://<galaxy_url>:80" nor "https://<galaxy_url>:443" are accessible
+        logging.critical(f"failed to guess URL from \"{galaxy_url}\" - {e}")
+        exit(2)
+
+    # test network connection and successful authentification
+    try:
+        response = gi.make_get_request(galaxy_url + "/api/whoami")
+        content = json.loads(response.content)
+
+        # this request should not fail
+        if response.status_code != 200:
+            logging.critical(
+                f"connection test failed - got HTTP status \"{response.status_code}\" with message \"{content['err_msg']}\"")
+            exit(2)
+
+        logging.info(f"connection successful - logged in as user \"{content['username']}\"")
+
+    except Exception as e:
+        # if the network connection fails, GalaxyInstance will throw an exception
+        logging.critical(f"exception during connection test - \"{e}\"")
+        exit(2)
+
+    # test connection with a GET to /api/whoami
+
+    # TODO do the test
+    # resp = gi.make_get_request(galaxy_url+ "/api/whoami")
+    # resp = gi.make_get_request(galaxy_url + "/api/configuration?keys=allow_user_deletion").content
+    return gi
+
+
+def get_beacon_histories(gi: GalaxyInstance) -> List[str]:

```

Comment:
Should those shared functions live in a separate file and be used by both scripts?